src/foam/util/AddressUtil.java

```diff
@@ -112,18 +112,28 @@ public static String normalizeCountry(X x, String countryCode) {
       return countryCode;
     }
     
-    String normalizedCountry = countryCode;
+    String[] normalizedCountry = { countryCode };
 
     DAO countryDAO = (DAO) x.get("countryDAO");
-    Country country = (Country) countryDAO.find(OR(
-      EQ(Country.ISO31661CODE, countryCode),
-      EQ(Country.CODE, countryCode)
-    ));
-
-    if ( country != null ) {
-      normalizedCountry = country.getCode();
-    }
+    countryDAO
+      .where(OR(
+        EQ(Country.ISO31661CODE, countryCode),
+        EQ(Country.CODE, countryCode),
+        STARTS_WITH_IC(Country.NAME, countryCode)
+      ))
+      .select(new AbstractSink() {
+        public void put(Object o, Detachable d) {
+          Country country = (Country) o;
+          if (
+            country.getIso31661Code().equals(countryCode) ||
```

Comment:
Why are you matching this twice? Once in the where condition above and again inside the sink?

Reply:
When STARTS_WITH was true, we need to another check with full name inside the sink. But in other cases (ISOCODE or CODE matched), we just want to quit early before we check against the full name inside the sink.